src/fstar/FStar.Universal.fs

```diff
@@ -139,9 +139,10 @@ let init_env deps : TcEnv.env =
 (* Interactive mode: checking a fragment of a code                     *)
 (***********************************************************************)
 let tc_one_fragment curmod (env:TcEnv.env_t) frag =
+  let file_from_env env = Range.file_of_range (TcEnv.get_range env) in
   let acceptable_mod_name modul =
     (* Interface is sent as the first chunk, so we must allow repeating the same module. *)
-    Parser.Dep.lowercase_module_name (List.hd (Options.file_list ())) =
+    Parser.Dep.lowercase_module_name (file_from_env env) =
```

Comment:
Can you please explain this change too, why does the previous code not work for LSP? (A comment in the code would be useful.)

Reply:
Lsp is started with `fstar.exe --lsp` without providing a single file in `Options.file_list ()`, while Interactive is started with `fstar.exe --ide filename.fst` providing a single file, making `hd` yield the desired result. I'll include a comment.